File: ProjetoAngular/BackEnd/Models/Livro.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackEnd.Models
{
    public class Livro
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int id { get; set; }

        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        [Column(TypeName = "varchar(70)")]
        public string nome { get; set; }

        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        public double edicao { get; set; }

        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        public int pagina { get; set; }

        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        [Column(TypeName = "varchar(20)")]
        public string tipoCapa { get; set; }

        [DisplayFormat(DataFormatString = "{0:dd-MM-yyyy}")]
        [DataType(DataType.Date)]
        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        public DateTime dataPublicacao { get; set; }

        [ForeignKey("Segmento")]
        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        public int segmentoId { get; set; }
        public Segmento segmento { get; set; }

        [ForeignKey("Marca")]
        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        public int marcaId { get; set; }
        public Marca marca { get; set; }

        [ForeignKey("Autor")]
        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        public int autorId { get; set; }
        public Autor autor { get; set; }

        [ForeignKey("Classificacao")]
        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        public int classificacaoId { get; set; }
        public Classificacao classificacao { get; set; }

        [ForeignKey("Genero")]
        [Required(ErrorMessage = "Campo {0} é Obrigatório")]
        public int generoId { get; set; }
        public Genero genero { get; set; }

        [Required]
        [RegularExpression(@".*\.(gif|jpe?g|bmp|png)$", ErrorMessage = "Não é uma imagem válida. (gif, jpg, jpeg, bmp ou png)")]
        public string imagemURL { get; set; }

        public IEnumerable<CompraItem> compraItems { get; set; }

        public Livro(int id, string nome, double edicao, int pagina, string tipoCapa, DateTime dataPublicacao, int segmentoId, int marcaId, int autorId, int classificacaoId, int generoId, string imagemURL)
        {
            this.id = id;
            this.nome = nome;
            this.edicao = edicao;
            this.pagina = pagina;
            this.tipoCapa = tipoCapa;
            this.dataPublicacao = dataPublicacao;
            this.segmentoId = segmentoId;
            this.marcaId = marcaId;
            this.autorId = autorId;
            this.classificacaoId = classificacaoId;
            this.generoId = generoId;
            this.imagemURL = imagemURL;
        }
    }
}

File: ProjetoAngular/LivrariaFrontEnd/src/app/models/Livro.ts
import { Autor } from "./Autor";
import { Classificacao } from "./Classificacao";
import { Genero } from "./Genero";
import { Marca } from "./Marca";
import { Segmento } from "./Segmento";

export class Livro {
    id: number;
    nome: string;
    edicao: number;
    pagina: number;
    tipoCapa: string;
    dataPublicacao: Date;
    segmentoId: number;
    segmento: Segmento;
    marcaId: number;
    marca: Marca;
    autorId: number;
    autor: Autor;
    classificacaoId: number;
    classificacao: Classificacao;
    generoId: number;
    genero: Genero;
    imagemURL: string;


    constructor() {
        this.id = 0;
        this.nome = '';
        this.edicao = 0;
        this.pagina = 0;
        this.tipoCapa = '';
        this.dataPublicacao = new Date();
        this.segmentoId = 0;
        this.segmento = new Segmento();
        this.marcaId = 0;
        this.marca = new Marca();
        this.autorId = 0;
        this.autor = new Autor();
        this.classificacaoId = 0;
        this.classificacao = new Classificacao();
        this.generoId = 0;
        this.genero = new Genero();
        this.imagemURL = '';
    }
}
